controller/src/test/java/io/pravega/controller/server/eventProcessor/SerializedRequestHandlerTest.java

```diff
@@ -0,0 +1,174 @@
+
+        s1e3.complete();
+
+        stream1Queue = requestHandler.getEventQueueForKey(getKeyForStream("scope", "stream1"));
+
+        // verify that no processing is complete
+        assertTrue(stream1Queue.size() >= 2);
+        assertTrue(stream1Queue.stream().noneMatch(x -> x.getRight().isDone()));
+        collect = stream1Queue.stream().map(x -> x.getLeft().getNumber()).collect(Collectors.toList());
+        assertTrue(collect.indexOf(2) < collect.indexOf(3));
+
+        // post 3 work for stream2
+        TestEvent s2e1 = new TestEvent("scope", "stream2", 1);
+        CompletableFuture<Void> s2p1 = requestHandler.process(s2e1, writer);
+        TestEvent s2e2 = new TestEvent("scope", "stream2", 2);
+        CompletableFuture<Void> s2p2 = requestHandler.process(s2e2, writer);
+        TestEvent s2e3 = new TestEvent("scope", "stream2", 3);
+        CompletableFuture<Void> s2p3 = requestHandler.process(s2e3, writer);
+
+        List<Pair<TestEvent, CompletableFuture<Void>>> stream2Queue = requestHandler.getEventQueueForKey(getKeyForStream("scope", "stream1"));
+        assertTrue(stream2Queue.size() >= 2);
+        assertTrue(stream2Queue.stream().noneMatch(x -> x.getRight().isDone()));
+        collect = stream2Queue.stream().map(x -> x.getLeft().getNumber()).collect(Collectors.toList());
+        assertTrue(collect.indexOf(2) < collect.indexOf(3));
+
+        s1e1.complete();
+        FutureHelpers.await(s1p1);
+
+        stream1Queue = requestHandler.getEventQueueForKey(getKeyForStream("scope", "stream1"));
+        assertTrue(stream1Queue.size() >= 1);
+        assertTrue(stream1Queue.stream().noneMatch(x -> x.getRight().isDone()));
+        collect = stream1Queue.stream().map(x -> x.getLeft().getNumber()).collect(Collectors.toList());
+        assertTrue(collect.contains(3));
+
+        // now make sure that we have concurrently run for two streams
+        s2e1.complete();
+        FutureHelpers.await(s2p1);
+
+        stream2Queue = requestHandler.getEventQueueForKey(getKeyForStream("scope", "stream1"));
+        assertTrue(stream2Queue.size() >= 1);
+        assertTrue(stream2Queue.stream().noneMatch(x -> x.getRight().isDone()));
+        collect = stream2Queue.stream().map(x -> x.getLeft().getNumber()).collect(Collectors.toList());
+        assertTrue(collect.contains(3));
+
+        // now complete all processing
+        s2e2.complete();
+        FutureHelpers.await(s2p2);
+
+        s2e3.complete();
+
+        s1e2.complete();
+        FutureHelpers.await(s1p2);
+
+        FutureHelpers.await(s1p3);
+        FutureHelpers.await(s2p3);
+
+        assertTrue(orderOfProcessing.get(s1e1.getKey()).get(0) == 1 &&
+                orderOfProcessing.get(s1e1.getKey()).get(1) == 2 &&
+                orderOfProcessing.get(s1e1.getKey()).get(2) == 3);
+        assertTrue(orderOfProcessing.get(s2e1.getKey()).get(0) == 1 &&
+                orderOfProcessing.get(s1e1.getKey()).get(1) == 2 &&
```

Comment:
should it be `s2e1.getKey()` in this line and the following?

Reply:
fixed